### PR TITLE
storage/{mysql,postgres}: add MerkleLeafHash index

### PR DESCRIPTION
**Pull request by daviddrysdale**
Requests for inclusion proofs need to look up leaves by their
Merkle leaf hash, but there wasn't an index for this so it
would involve a full table scan.

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

**Comment by RJPercival** [comment]
I've tested this on a MySQL-backed Trillian instance and it does indeed massively accelerate the query required to answer inclusion proof requests.

File: storage/mysql/storage.sql [hunk 1]
```diff
@@ -111,6 +111,10 @@ CREATE TABLE IF NOT EXISTS SequencedLeafData(
   FOREIGN KEY(TreeId, LeafIdentityHash) REFERENCES LeafData(TreeId, LeafIdentityHash) ON DELETE CASCADE
 );
 
+
+CREATE INDEX SequencedLeafMerkleIdx
+  ON SequencedLeafData(TreeId, MerkleLeafHash);
```
Comment (RJPercival) [inline storage/mysql/storage.sql]: Why not put this in the `CREATE TABLE` definition?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,11 @@
 
 Not yet released; provisionally v2.0.0 (may change).
 
+### Database Schema
+
+This version includes a change to the MySQL and Postgres database schemas
+to add an index on the `SequencedLeafData` table.
```
[the remark this answers is not in the document]
Comment (daviddrysdale) [reply CHANGELOG.md]: Done.

File: storage/mysql/storage.sql (hunk 1 above)
Comment (daviddrysdale) [reply storage/mysql/storage.sql]: That would involve me knowing more SQL syntax than I do...

Comment (daviddrysdale) [reply storage/mysql/storage.sql]: I'm going to leave as-is for consistency with other indices in there.

**Pull request merged**